### PR TITLE
Adding new Alpha Model. DualThrust

### PR DESCRIPTION
**Pull request by IlshatGaripov**
#### Description
This commit publishes new alpha model into QuantConnect.Algorithm.Framework.Alphas namespace.
Written in C#.
Alpha model implements a widely known concept to generate trading singnals called Dual Thrust. https://www.quantconnect.com/tutorials/strategy-library/dual-thrust-trading-algorithm

#### Related Issue
NA

#### Motivation and Context
Add more alphas to an opensource. This model implements the basic idea and can be use in conjunction with other alpha models.

#### Requires Documentation Change
NA

#### How Has This Been Tested?
I have run it locally on custom data. It does work fine.

#### Types of changes
* [ ]  Bug fix (non-breaking change which fixes an issue)
* [X ] New feature (non-breaking change which adds functionality)
* [ ]  Breaking change (fix or feature that would cause existing functionality to change)
* [ ]  Non-functional change (xml comments/documentation/etc)

#### Checklist:
* [ X] My code follows the code style of this project.
* [ X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
* [ ]  I have added tests to cover my changes.
* [ ]  All new and existing tests passed.
* [X ] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Framework/Alphas/DualThrustAlphaModel.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
+﻿using QuantConnect.Data;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (jaredbroad) [inline Algorithm.Framework/Alphas/DualThrustAlphaModel.cs]: Some of these could be removed I think

Comment (jaredbroad) [reply Algorithm.Framework/Alphas/DualThrustAlphaModel.cs]: Please follow this pattern the Algorithm.Language folder has algorithms; see following for examples:
https://github.com/QuantConnect/Lean/pull/2932
https://github.com/QuantConnect/Lean/pull/2931
https://github.com/QuantConnect/Lean/pull/2930

**Comment by jaredbroad** [comment]
Closing pending updates from review.

**Comment by IlshatGaripov** [comment]
@jaredbroad thank you Jared, will re submit. sorry busy days and couldn't yet find good parameters, even for demonstration. thanks,..

**Pull request closed without merging**